corva/configuration.py

```diff
@@ -0,0 +1,40 @@
+from os import getenv
+from typing import Optional
+
+from pydantic import BaseSettings
+
+
+class Settings(BaseSettings):
+    # api
+    API_ROOT_URL: Optional[str] = None
+    DATA_API_ROOT_URL: Optional[str] = None
+    API_KEY: Optional[str] = None
+
+    # cache
+    CACHE_URL: Optional[str] = None
+
+    # logger
+    LOG_LEVEL: str = 'WARN'
+
+    # misc
+    APP_KEY: Optional[str] = None  # <provider>.<app-name-with-dashes>
+
+    @property
+    def APP_NAME(self) -> str:
+        if app_name := getenv('APP_NAME') is not None:
+            return app_name
+
+        app_name_with_dashes = self.APP_KEY.split('.')[1]
```

Comment:
This could cause an index error if the app key is mangled. Also, this should probably limit the splitting to 1 result.

Reply:
true, but this is our responsibility to set correct env vars